### PR TITLE
Add cases to cover scsi host device passthough

### PR DESCRIPTION
**Pull request by lento-sun**
Hotplug/unplug, coldplug or edit xml will be used to add scsi
hostdev, and all hostdev attrs will be covered.

Signed-off-by: Yi Sun <yisun@redhat.com>

### CODE REVIEW

**Comment by lento-sun** [comment]
This PR is still work in progress, submit it to make sure some involved cases can be run. 
But more verification point will be added and script will be tidy up then, so pls do not review it for now.

**Comment by lento-sun** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/1186

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+            with open(chardev_unpriv_path, 'r') as f:
+                chardev_unpriv_value = f.read().strip()
+            logging.debug("blkdev unpriv_sgio:%s\nchardev unpriv_sgio:%s",
+                          blkdev_unpriv_value, chardev_unpriv_value)
+            if ((not blkdev_unpriv_value or not chardev_unpriv_value) or
+                    (blkdev_unpriv_value != chardev_unpriv_value)):
+                test.error("unpriv_sgio values are incorrect under block "
+                           "and char folders.")
+            return blkdev_unpriv_value
+        except Exception, detail:
```
Comment (hsmj1412) [inline libvirt/tests/src/libvirt_scsi_hostdev.py]: except Exception as detail:

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+                          status, output)
+            if readonly:
+                for ro_kw in readonly_keywords:
+                    if ro_kw in str(output).lower():
+                        return True
+                logging.error("Hostdev set with 'readonly'. "
+                              "But still can be operated.")
+                return False
+            return status == 0
+        except (remote.LoginError, virt_vm.VMError, aexpect.ShellError), err:
```
Comment (hsmj1412) [inline libvirt/tests/src/libvirt_scsi_hostdev.py]: except (remote.LoginError, virt_vm.VMError, aexpect.ShellError) as err:

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            new_partitions = libvirt.get_parts_list(session)
+            logging.debug("new partitions are: %s", new_partitions)
+            added_partitions = list(set(new_partitions).difference(set(old_partitions)))
+            session.close()
+            if not added_partitions:
+                logging.debug("No new partitions found in vm.")
+            else:
+                logging.debug("Newly added partition(s) is: %s", added_partitions)
+            return added_partitions
+        except (remote.LoginError, virt_vm.VMError, aexpect.ShellError), err:
```
Comment (hsmj1412) [inline libvirt/tests/src/libvirt_scsi_hostdev.py]: except (remote.LoginError, virt_vm.VMError, aexpect.ShellError) as err:

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                                         flagstr=attach_options,
+                                         ignore_status=True, debug=True)
+            libvirt.check_exit_status(result, status_error & hotplug)
+        if coldplug:
+            vm.destroy(gracefully=False)
+            result = virsh.start(vm_name, ignore_status=True, debug=True)
+            libvirt.check_exit_status(result, status_error & coldplug)
+        if not status_error:
+            vm.wait_for_login().close()
+            time.sleep(10)
```
Comment (kylazhang) [inline libvirt/tests/src/libvirt_scsi_hostdev.py]: remove the sleep here and maybe increase the wait_for timeout

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+                    if not check_disk_io(vm2, vm2_new_disk):
+                        test.fail("Testing shareable device, got unexpected result "
+                                  "when operate the newly added disk in the second vm.")
+
+            # Detach the disk from vm.
+            for dev_num in range(device_num):
+                result = virsh.detach_device(vm_name, hostdev_xmls[dev_num].xml,
+                                             flagstr=attach_options,
+                                             ignore_status=True, debug=True)
+                libvirt.check_exit_status(result)
```
Comment (kylazhang) [inline libvirt/tests/src/libvirt_scsi_hostdev.py]: If only check the detach_device result, no need to use check_exit_status, just setting ignore_status=False will raise error directly. Please also check other similar part in the code for the usage like this.

File: libvirt/tests/src/libvirt_scsi_hostdev.py [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+        """
+        Get new disks in vm after hostdev plug.
+        :param vm: The vm to be checked.
+        :param old_partitions: Already existing partitions in vm.
+        :return: New disks/partitions in vm.
+        """
+        try:
+            session = vm.wait_for_login()
+            if platform.platform().count('ppc64'):
+                time.sleep(10)
```
[the remark this answers is not in the document]
Comment (lento-sun) [reply libvirt/tests/src/libvirt_scsi_hostdev.py]: I'll remove this one, it was a ppc specific issue without known reason. If something wrong happens on ppc again, related owner can make a better fix about this on ppc

File: libvirt/tests/src/libvirt_scsi_hostdev.py (hunk 4 above)
Comment (lento-sun) [reply libvirt/tests/src/libvirt_scsi_hostdev.py]: I'll remove this for now

File: libvirt/tests/src/libvirt_scsi_hostdev.py (hunk 5 above)
Comment (lento-sun) [reply libvirt/tests/src/libvirt_scsi_hostdev.py]: will update

**Comment by lento-sun** [comment]
@kylazhang thx for te review, updated accordingly

**Pull request merged**